### PR TITLE
Fix hanging IteratorExecutor. HiveDataset will skip bad tables instea…

### PR DESCRIPTION
**Pull request by ibuenros**
…d of failing the entire job.

This PR fixes two issues:
* IteratorExecutor does not shutdown executor on a failure, making job hang.
* HiveDatasetFinder fails if a single table is bad. Fix will skip and report bad tables.

### CODE REVIEW

**Comment by ibuenros** [comment]
@ydai1124 can you review? Thanks.

File: gobblin-utility/src/main/java/gobblin/util/executors/IteratorExecutor.java [hunk 1]
```diff
@@ -59,29 +59,32 @@ public IteratorExecutor(Iterator<Callable<T>> runnableIterator, int numThreads,
...
-    while (activeTasks > 0) {
-      this.completionService.take();
-      activeTasks--;
-    }
 
-    this.completionService.poll();
-
-    ExecutorsUtils.shutdownExecutorService(this.executor, Optional.of(log), 10, TimeUnit.SECONDS);
-    this.executed = true;
+      this.completionService.poll();
```
Comment (ydai1124) [inline gobblin-utility/src/main/java/gobblin/util/executors/IteratorExecutor.java]: Is `poll` necessary here? `take` seems have taken care of all completion service?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java [hunk 2]
```diff
@@ -134,20 +157,32 @@ public boolean apply(String table) {
...
-          Table table = client.get().getTable(dbAndTable.getDb(), dbAndTable.getTable());
-          return createHiveDataset(table);
-        } catch (IOException | TException ioe) {
-          throw new RuntimeException(ioe);
+      protected HiveDataset computeNext() {
+        while (this.tables.hasNext()) {
+          DbAndTable dbAndTable = this.tables.next();
+          try (AutoReturnableObject<IMetaStoreClient> client = HiveDatasetFinder.this.clientPool.getClient()) {
+            Table table = client.get().getTable(dbAndTable.getDb(), dbAndTable.getTable());
+            if (HiveDatasetFinder.this.eventSubmitter.isPresent()) {
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java]: nit: create a method to optionally submit events which checks if `eventSubmitter` is present. So you don't need to have this `if` blocks twice.

**Comment by ibuenros** [comment]
@ydai1124 addressed comments.

**Comment by ydai1124** [comment]
LGTM

**Pull request merged**